core/src/main/java/software/amazon/awssdk/util/DateUtils.java

```diff
@@ -256,15 +241,34 @@ public static String formatServiceSpecificDate(Date date) {
                         .toPlainString();
     }
 
+    /**
+     * Formats the give date object into an AWS Service format.
+     */
+    public static String formatServiceSpecificDate(Instant instant) {
+        if (instant == null) {
+            return null;
+        }
+        BigDecimal dateValue = BigDecimal.valueOf(instant.toEpochMilli());
+        return dateValue.scaleByPowerOfTen(0 - AWS_DATE_MILLI_SECOND_PRECISION)
+                   .toPlainString();
+    }
+
+    @Deprecated
+    @ReviewBeforeRelease("not used.")
     public static Date cloneDate(Date date) {
         return date == null ? null : new Date(date.getTime());
     }
 
     /**
      * Returns the number of days since epoch with respect to the given number
      * of milliseconds since epoch.
+     * @deprecated use {@link Duration#ofMillis(long)} and {@link Duration#toDays()}.
      */
-    public static long numberOfDaysSinceEpoch(long milliSinceEpoch) {
-        return TimeUnit.MILLISECONDS.toDays(milliSinceEpoch);
+    public static Duration durationSinceEpoch(long milliSinceEpoch) {
+        return Duration.ofMillis(milliSinceEpoch);
+    }
+
+    private static Instant parseInstant(String dateString, DateTimeFormatter formatter) {
+        return ZonedDateTime.parse(dateString, formatter).toInstant();
```

Comment:
Should be able to set the zone on the formatter directly using `withZone` instead of going through `ZonedDateTime`